internal/usecase/interactor/user.go

```diff
@@ -241,6 +169,109 @@ func (i *User) Signup(ctx context.Context, inp interfaces.SignupParam) (u *user.
 	return u, team, nil
 }
 
+func (i *User) authSystemSignup(ctx context.Context, inp interfaces.SignupParam, tx repo.Tx) (repo.Tx, string, string, *user.User, *user.Team, error) {
+	if *inp.Name == "" {
+		return nil, "", "", nil, nil, interfaces.ErrSignupInvalidName
+	}
+	if _, err := mail.ParseAddress(*inp.Email); err != nil {
+		return nil, "", "", nil, nil, interfaces.ErrInvalidUserEmail
+	}
+	if *inp.Password == "" {
+		return nil, "", "", nil, nil, interfaces.ErrSignupInvalidPassword
+	}
+
+	tx, err := i.transaction.Begin()
+	if err != nil {
+		return nil, "", "", nil, nil, err
+	}
+	defer func() {
+		if err2 := tx.End(ctx); err == nil && err2 != nil {
+			err = err2
+		}
+	}()
```

Comment:
There are duplication of this and line 228. 
- Begin transaction outside this function.
- There is no need to accept and return `repo.Transaction` and `repo.Tx`. 

Fix `auth0Signup` also.

Reply:
I wanted to start the transaction just before the query code

Reply:
Current:

pseudo code

```
if auth0 {
  start tx // duplicated
  auth0Signup()
} else if authSystem {
  start tx // duplicated
  authSystemSignup()
}
```

Goal:

pseudo code
```
if !auth0 && !signup {
  return
}
start tx
if auth0 {
  auth0Signup()
} else if authSystem {
  authSystemSignup()
}
```